### PR TITLE
Migrate to prop-types and create-react-class packages

### PR DESCRIPTION
**Pull request by agirton**
In React 15.5 they are starting the process to migrate away from having `PropTypes` and `createClass` under the main `React` object. This PR updates all of the related files needed to use the new packages.

I ran all of the unit tests and also tested the examples app after the changes. Both had no errors/issues.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,8 +11,10 @@
     "url": "https://github.com/JedWatson/react-select.git"
   },
   "dependencies": {
+    "create-react-class": "15.5.1",
     "classnames": "^2.2.4",
-    "react-input-autosize": "^1.1.0"
+    "react-input-autosize": "^1.1.0",
```
Comment (merlinpatt) [inline package.json]: This dependency will need to be updated as well. I've created a PR [here](https://github.com/JedWatson/react-input-autosize/pull/79) to do so

Comment (Betree) [inline package.json]: The lib has been updated, fixed version is [1.1.1](https://github.com/JedWatson/react-input-autosize/pull/79/files/6bb9330dcf473d1252d67988ef2cc069aa297764)

**Comment by DimaRGB** [comment]
Have some problems in CI, `Error: Cannot find module 'object-assign'`

File: package.json (hunk 1 above)
Comment (bmv437) [inline package.json]: react-autosize-input still needs to publish version 1.1.1 to npm

Comment (MichaelDeBoey) [inline package.json]: `react-autosize-input` has released `v1.1.3` :-)

**Comment by malimichael** [comment]
When can we expect this fix? It's my only package still throwing the warnings

**Comment by agirton** [comment]
@JedWatson this should be good to go.

**Comment by tomasztomys** [comment]
When can we expect release new version with this PR?

**Comment by ddrozdov** [comment]
@JedWatson would you merge this please?

**Comment by marcellodesales** [comment]
Just got this error :( Can't load my component... :(

**Comment by bmv437** [comment]
@marcellodesales technically in react 15.5 this is just a warning and shouldn't prevent the component from loading. When react 16 is released, it will be a breaking change.

**Comment by laalex** [comment]
👍

**Comment by Psiablo** [comment]
Merge @JedWatson?

**Comment by bertho-zero** [comment]
You can use `npm i JedWatson/react-select#pull/1658/head` waiting merge...

**Comment by FalloutX** [comment]
I am still getting when running the package from npm. Do I need to install the package from git? 

```
warning.js:36 Warning: AsyncCreatableSelect: React.createClass is deprecated and will be removed in version 16. Use plain JavaScript classes instead. If you're not yet ready to migrate, create-react-class is available on npm as a drop-in replacement.
```

**Comment by DimaRGB** [comment]
@FalloutX need to merge this pr.

**Comment by JedWatson** [comment]
Looks great @agirton, thanks and sorry for the delay merging (I've been away for personal reasons)

Will get this published asap so everyone stops getting annoyed by those warnings!

**Pull request merged**